### PR TITLE
FEAT: UPDATE THE ALGORITHIM FOR CALCULATION OF AVERAGE RATING ON INTERVIEW FEEDBACK AND UPDATE IT ON JOB APPLICANT DOCTYPE

### PR DESCRIPTION
**Pull request by NoBoneZ**
## Is this a Feature, Chore or Bug?
- [X] Feature
- [] Chore
- [] Bug

## Clearly and concisely describe the feature, chore or bug.
The algorithim for the calculation of the interview feedback needed to be updated, and also the job applicant document to be updated also, and to make the score and rating field on the child table in the interview feedback not mandatory

## Analysis and design (optional)
Analyse and attach the design documentation

## Solution description
Describe your code changes in detail for reviewers.
made the neccessary fields not mandatory, updated the algorithim to include the questions and also update the job applicant document

## Is there a business logic within a doctype?
    - [] Yes
    - [x] No

## Output screenshots (optional)
Post the output screenshots, if a UI is affected or added due to this feature.
[Screencast from 03-12-2024 16:21:33.webm](https://github.com/user-attachments/assets/38bef4f2-cc6c-48f7-897d-bf2a2b0a4b0f)
[Screencast from 03-12-2024 16:29:25.webm](https://github.com/user-attachments/assets/93eae01d-3486-4116-9570-419f8d31643a)

## Areas affected and ensured
List out the areas affected by your code changes.
Job applicant
Interview Feedback

## Is there any existing behavior change of other features due to this code change?
Mention Yes or No. If Yes, provide the appropriate explanation.

## Did you test with the following dataset?
- [x] Existing Data
- [x] New Data

## Was child table created?
    - [] is attachment required?
        did you test attachment
## Did you delete custom field?
    - [] Yes
    - [x] No
        If yes, did you write a delete patch?

## Is patch required?
- [] Yes
- [x] No
    ## Was the patch test?

## Which browser(s) did you use for testing?
  - [x] Chrome
  - [] Safari
  - [] Firefox

### CODE REVIEW

File: one_fm/fixtures/property_setter.json [hunk 1]
```diff
@@ -5925,271 +5893,1023 @@
...
+  "doctype": "Property Setter",
+  "doctype_or_field": "DocType",
+  "field_name": null,
+  "is_system_generated": 0,
+  "modified": "2024-12-03 15:39:41.986200",
+  "module": null,
+  "name": "Employee-main-field_order",
+  "property": "field_order",
+  "property_type": "Data",
+  "row_name": null,
```
Comment (pjamsheer) [inline one_fm/fixtures/property_setter.json]: This and lot changes not relevant to this PR

File: one_fm/overrides/interview_feedback.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+            divisor = (1 if not all([self.skill_assessment, self.interview_question_assessment]) else 2)
+            
+            self.average_rating = flt((skill_assessment_avg + interview_assessment_avg) / divisor) / 5
+        except Exception as e:
+            frappe.log_error(frappe.get_traceback(), f"Error calculating average rating: {e}")
+            self.average_rating = 0
+    
+    def after_insert(self):
+        frappe.db.set_value("Job Applicant", self.job_applicant, "custom_interview_feedback_rating", self.average_rating)
```
Comment (pjamsheer) [inline one_fm/overrides/interview_feedback.py]: Interview feedback is submittable, what happened if the user update the feedback after saving the feedback

File: one_fm/one_fm/custom/job_applicant.json [hunk 3]
```diff
@@ -1,5 +1,137 @@
...
+   "columns": 0,
+   "creation": "2024-12-03 13:14:45.325999",
+   "default": null,
+   "depends_on": null,
+   "description": null,
+   "docstatus": 0,
+   "dt": "Job Applicant",
+   "fetch_from": null,
+   "fetch_if_empty": 0,
+   "fieldname": "custom_interview_feedback_rating",
```
Comment (pjamsheer) [inline one_fm/one_fm/custom/job_applicant.json]: What is the use of this field, we can update the value to `Applicant Rating` field in Job Applicant

File: one_fm/one_fm/custom/job_applicant.json [hunk 4]
```diff
@@ -1,5 +1,137 @@
...
+   "is_system_generated": 0,
+   "is_virtual": 0,
+   "label": "Interview Feedback",
+   "length": 0,
+   "link_filters": null,
+   "mandatory_depends_on": null,
+   "modified": "2024-12-03 13:14:45.064791",
+   "modified_by": "Administrator",
+   "module": null,
+   "name": "Job Applicant-custom_interview_feedback",
```
Comment (pjamsheer) [inline one_fm/one_fm/custom/job_applicant.json]: In Job applicant there is an inbuilt summary section
Screenshot 2024-12-03 at 9 45 28 PM

Comment (NoBoneZ) [reply one_fm/one_fm/custom/job_applicant.json]: @pjamsheer , Yes, but that might take some time, and the need it tomorrow, i could work on it after they are done with the interviews tomorrow

File: one_fm/one_fm/custom/job_applicant.json (hunk 3 above)
Comment (NoBoneZ) [reply one_fm/one_fm/custom/job_applicant.json]: This is the rating field that gets updated, when the interfiew feedback is saved

File: one_fm/overrides/interview_feedback.py (hunk 2 above)
Comment (NoBoneZ) [reply one_fm/overrides/interview_feedback.py]: I could change it to before_save, then

File: one_fm/one_fm/custom/job_applicant.json (hunk 4 above)
Comment (NoBoneZ) [reply one_fm/one_fm/custom/job_applicant.json]: or rather, if i use the applicant rating, there wont be need for that anymore

File: one_fm/one_fm/custom/job_applicant.json (hunk 3 above)
Comment (NoBoneZ) [reply one_fm/one_fm/custom/job_applicant.json]: @pjamsheer  i will update this to use the applicant rating

**Review by pjamsheer — APPROVED** [review]

**Pull request merged**